kit/ChildSession.cpp

```diff
@@ -1639,6 +1644,28 @@ bool ChildSession::mouseEvent(const StringVector& tokens,
     return true;
 }
 
+bool ChildSession::readOnlyClickEvent(const StringVector& tokens)
+{
+    getLOKitDocument()->setView(_viewId);
+    int x = std::atoi(tokens[1].substr(2).data());
+    int y = std::atoi(tokens[2].substr(2).data());
+
+    char *test = getLOKitDocument()->hyperlinkInfoAtPosition(x, y);
+    std::string result(test);
+    std::free(test);
+
+    if (!result.empty())
+    {
+        std::string message = "readonlyhyperlinkclicked: { \"link\": \"%%LINK_HERE%%\", \"position\": { \"x\": %%X_HERE%%, \"y\": %%Y_HERE%% } }";
```

Comment:
Much simpler, safer, faster, better, to simply use `std::ostringstream` and stream/concatenate the values directly into it.

Creating a template and then doing search/replace 3 times, with `const_cast` is not right.